pkg/securitycontext/util.go

```diff
@@ -61,12 +62,13 @@ func DetermineEffectiveSecurityContext(pod *v1.Pod, container *v1.Container) *v1
 		return containerSc
 	}
 
```

Comment:
`DetermineEffectiveSecurityContext` returns right away if the checks [above](https://github.com/kubernetes/kubernetes/blob/3365ed5625797cfaafbdab368e901e9f4739329a/pkg/securitycontext/util.go#L57-L62) are satisfied. In that case, if the `effectiveSc` or `containerSc` happens to have the members of a security context that do not apply to Windows, they will continue to be returned (and will not be stripped as below).

Since the desired action is specific to Windows, have you considered cleaning up the security contexts in the Windows [caller](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/kuberuntime_container_windows.go#L128)? That would:
1. Allow the early/optimized exit paths above to continue to work.
2. Allow `DetermineEffectiveSecurityContext` to be invoked from a cluster-wide controller context (if needed in the future) rather than the node-specific kubelet context. The `runtime.GOOS != "windows"` checks assumes `DetermineEffectiveSecurityContext` will never be invoked from a cluster-wide controller context.

Reply:
Doing this work in the windows specific code makes sense. We have run into a few other places where this has come up with security context: https://github.com/kubernetes/kubernetes/pull/92355. In the future there could be a place where some of the these attributes might be available if a form of privileged containers becomes available.

The move to doing the checks in windows specific code makes the resulting logic here much simpler over adding complex if statements which will become hard to maintain and easily introduce bugs.